lib/iris/coords.py

```diff
@@ -1240,13 +1240,20 @@ def xml_element(self, doc):
 
     def _xml_id(self):
         # Returns a consistent, unique string identifier for this coordinate.
-        unique_value = (self.standard_name, self.long_name, self.units,
-                        tuple(sorted(self.attributes.items())),
-                        self.coord_system)
+        unique_value = b''
```

Comment:
The trouble is the `tuple` wherein any contained Unicode strings might get a `u` prefix when on Python 2, but not Python 3. So I went for a somewhat more verbose, but more explicit method.